esp-hal/build.rs

```diff
@@ -80,23 +80,27 @@ fn main() -> Result<(), Box<dyn Error>> {
         &[
             (
                 "place-spi-driver-in-ram",
-                Value::Bool(false),
                 "Places the SPI driver in RAM for better performance",
+                Value::Bool(false),
+                None
```

Comment:
This is a purely theoretical question, but what does it mean to set a, say, range requirement for a boolean? I guess what I want to say is, wouldn't it be better to describe the validation rule as part of the value type, as `Value::XY(default, validation_rule)`?

Reply:
The built-in validators already perform type checking and return an error if applied to an invalid type. Given we are the intended users of this functionality I thought this was a reasonable approach.